cluster-management/setup/update-strategies/index.md

```diff
@@ -137,23 +171,47 @@ coreos:
   update:
     reboot-strategy: off
   units:
-    -
-      name: update-window.service
+    - name: update-window.service
       runtime: true
       content: |
         [Unit]
         Description=Reboot if an update has been downloaded
 
         [Service]
-        ExecStart=/usr/bin/bash -c 'if update_engine_client -status | grep NEED_REBOOT; then locksmithctl reboot; fi' 
-    -
-      name: update-window.timer
+        ExecStart=/usr/bin/bash /opt/bin/update-window.sh 
+    - name: update-window.timer
       runtime: true
       command: start
       content: |
         [Unit]
-        Description=Reboot if needed at 05:00 daily
+        Description=Reboot if needed at 05:10,05:30,05:50, and 06:10,06:30,06:50.
 
         [Timer]
-        OnCalendar=*-*-* 05:00:00
+        OnCalendar=*-*-* 05,06:10/20:00
+        
+write_files:
+  - path: /opt/bin/update-window.sh
+    content: |
+        #!/bin/bash
+        # Randomly delay reboot between 0 and 3600 seconds.
+        # If etcd is not running, simply reboot, otherwise use locksmithctl reboot to get reboot lock.
+        delay=$(/usr/bin/expr $RANDOM % 3600 )
+        rebootflag='NEED_REBOOT'
+        # For testing force reboot, uncomment the following line
```

Comment:
@crawford - remove the /opt/bin/update-window.sh installation from the cloud-config?